VL.Data/Services/SeedingService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Bogus;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using VL.Shared.Data;
+using VL.Shared.Interfaces;
+using VL.Shared.Models;
+
+namespace VL.Shared.Services
+{
+    public class SeedingService : ISeedingService
+    {
+        private readonly ApplicationDbContext _applicationDbContext;
+        private readonly Faker _faker;
+        private readonly UserManager<IdentityUser> _userManager;
+        private readonly IUserStore<IdentityUser> _userStore;
+        private readonly IUserEmailStore<IdentityUser> _emailStore;
+        private readonly RoleManager<IdentityRole> _roleManager;
+
+        public SeedingService(
+            ApplicationDbContext applicationDbContext,
+            UserManager<IdentityUser> userManager,
+            IUserStore<IdentityUser> userStore,
+            RoleManager<IdentityRole> roleManager)
+        {
+            _applicationDbContext = applicationDbContext;
+            _faker = new Faker
+            {
+                Random = new Randomizer(8675309)
+            };
+            _userManager = userManager;
+            _userStore = userStore;
+            _emailStore = GetEmailStore();
+            _roleManager = roleManager;
+        }
+
+        public async Task SeedBooksAsync(int count)
+        {
+            if (await _applicationDbContext.Book.AnyAsync())
+            {
+                return;
+            }
+
+            Randomizer.Seed = new Random(1338);
+            var books = new Faker<Book>()
+                .RuleFor(u => u.Title, (f, u) => f.Commerce.ProductName())
+                .RuleFor(u => u.Author, (f, u) => f.Person.FullName)
+                .RuleFor(u => u.ReleaseDate, f => f.Date.Past(_faker.Random.Number(1, 10)))
+                .RuleFor(u => u.CheckedOut, (f, u) => f.Random.Bool())
+                .RuleFor(u => u.Description, (f, u) => f.Commerce.ProductDescription())
+                .Generate(count);
+
+            await _applicationDbContext.AddRangeAsync(books);
+
+            await _applicationDbContext.SaveChangesAsync();
+        }
+
+        public async Task SeedRolesAsync()
+        {
+            if (await _applicationDbContext.Roles.AnyAsync())
+            {
+                return;
+            }
+
+            await _roleManager.CreateAsync(new IdentityRole(Roles.Librarian));
+            await _roleManager.CreateAsync(new IdentityRole(Roles.User));
+        }
+
+        public async Task SeedUsersAsync()
+        {
+            if (await _applicationDbContext.Users.AnyAsync())
+            {
+                return;
+            }
+
+            var user = CreateUser();
+
+            await _userStore.SetUserNameAsync(user, "user@vl.com", CancellationToken.None);
```

Comment:
Only _userManager and _roleManager is required here

Create a new instance of a class using the 'new' keyword